core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -1697,6 +1697,9 @@ public void doUpdateUserListOfRoleWithID(String roleName, String[] deletedUserID
             }
             throw new UserStoreException(msg, e);
         } catch (Exception e) {
+            if (e instanceof UserStoreException) {
+                throw (UserStoreException) e; // Re-throwing UserStoreException
+            }
```

Comment:
What if we use a separate catch block for UserStoreException and throw, instead of checking the instance type?

Reply:
fixed